diff --git a/bin/kafka-server-start b/bin/kafka-server-start
index 5c364fb..53a1a26 100755
--- a/bin/kafka-server-start
+++ b/bin/kafka-server-start
@@ -50,4 +50,4 @@ case $COMMAND in
     ;;
 esac
 
-exec $base_dir/kafka-run-class $EXTRA_ARGS kafka.Kafka $@
+exec $base_dir/kafka-run-class $EXTRA_ARGS io.confluent.support.metrics.SupportedKafka $@
diff --git a/bin/kafka-server-stop b/bin/kafka-server-stop
index cd8160c..b4755c3 100755
--- a/bin/kafka-server-stop
+++ b/bin/kafka-server-stop
@@ -13,4 +13,4 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-ps ax | grep -i 'kafka\.Kafka' | grep java | grep -v grep | awk '{print $1}' | xargs kill -SIGTERM
+ps ax | grep -i 'io\.confluent\.support\.metrics\.SupportedKafka' | grep java | grep -v grep | awk '{print $1}' | xargs kill -SIGTERM
diff --git a/bin/windows/kafka-server-start.bat b/bin/windows/kafka-server-start.bat
index ae56b48..aba999c 100644
--- a/bin/windows/kafka-server-start.bat
+++ b/bin/windows/kafka-server-start.bat
@@ -26,5 +26,5 @@ if exists %~dp0../../etc/kafka/log4j.properties (
 	set KAFKA_LOG4J_OPTS=-Dlog4j.configuration=file:%~dp0../../config/log4j.properties
 )
 set KAFKA_HEAP_OPTS=-Xmx1G -Xms1G
-%~dp0kafka-run-class.bat kafka.Kafka %*
+%~dp0kafka-run-class.bat io.confluent.support.metrics.SupportedKafka %*
 EndLocal
diff --git a/bin/windows/kafka-server-stop.bat b/bin/windows/kafka-server-stop.bat
index 676577c..85212c5 100644
--- a/bin/windows/kafka-server-stop.bat
+++ b/bin/windows/kafka-server-stop.bat
@@ -14,5 +14,5 @@ rem WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 rem See the License for the specific language governing permissions and
 rem limitations under the License.
 
-wmic process where (commandline like "%%kafka.Kafka%%" and not name="wmic.exe") delete
-rem ps ax | grep -i 'kafka.Kafka' | grep -v grep | awk '{print $1}' | xargs kill -SIGTERM
+wmic process where (commandline like "%%io.confluent.support.metrics.SupportedKafka%%" and not name="wmic.exe") delete
+rem ps ax | grep -i 'io\.confluent\.support\.metrics\.SupportedKafka' | grep java | grep -v grep | awk '{print $1}' | xargs kill -SIGTERM
diff --git a/config/server.properties b/config/server.properties
index 8595ec8..5a7bd91 100644
--- a/config/server.properties
+++ b/config/server.properties
@@ -121,3 +121,24 @@ zookeeper.connect=localhost:2181
 
 # Timeout in ms for connecting to zookeeper
 zookeeper.connection.timeout.ms=6000
+
+##################### Confluent Proactive Support ######################
+
+# If set to true, then the feature to collect and report support metrics
+# ("Metrics") is enabled.  If set to false, the feature is disabled.
+#
+confluent.support.metrics.enable=true
+
+# The customer ID under which support metrics will be collected and
+# reported.
+#
+# When the customer ID is set to "anonymous" (the default), then only a
+# reduced set of metrics is being collected and reported.
+#
+# Confluent customers
+# -------------------
+# If you are a Confluent customer, then you should replace the default
+# value with your actual Confluent customer ID.  Doing so will ensure
+# that additional support metrics will be collected and reported.
+#
+confluent.support.customer.id=anonymous
