diff --git a/bin/connect-distributed b/bin/connect-distributed
index 354c3cc..65c5f8b 100755
--- a/bin/connect-distributed
+++ b/bin/connect-distributed
@@ -17,7 +17,16 @@
 base_dir=$(dirname $0)
 
 if [ "x$KAFKA_LOG4J_OPTS" = "x" ]; then
-    export KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:$base_dir/../config/connect-log4j.properties"
+  LOG4J_CONFIG_NORMAL_INSTALL="/etc/kafka/connect-log4j.properties"
+  LOG4J_CONFIG_ZIP_INSTALL="$base_dir/../etc/kafka/connect-log4j.properties"
+  if [ -e "$LOG4J_CONFIG_NORMAL_INSTALL" ]; then # Normal install layout
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:${LOG4J_CONFIG_NORMAL_INSTALL}"
+  elif [ -e "${LOG4J_CONFIG_ZIP_INSTALL}" ]; then # Simple zip file layout
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:${LOG4J_CONFIG_ZIP_INSTALL}"
+  else # Fallback to normal default
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:$base_dir/../config/connect-log4j.properties"
+  fi
 fi
+export KAFKA_LOG4J_OPTS
 
 exec $(dirname $0)/kafka-run-class org.apache.kafka.connect.cli.ConnectDistributed "$@"
diff --git a/bin/connect-standalone b/bin/connect-standalone
index 85904f7..ca58fe6 100755
--- a/bin/connect-standalone
+++ b/bin/connect-standalone
@@ -17,7 +17,16 @@
 base_dir=$(dirname $0)
 
 if [ "x$KAFKA_LOG4J_OPTS" = "x" ]; then
-    export KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:$base_dir/../config/connect-log4j.properties"
+  LOG4J_CONFIG_NORMAL_INSTALL="/etc/kafka/connect-log4j.properties"
+  LOG4J_CONFIG_ZIP_INSTALL="$base_dir/../etc/kafka/connect-log4j.properties"
+  if [ -e "$LOG4J_CONFIG_NORMAL_INSTALL" ]; then # Normal install layout
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:${LOG4J_CONFIG_NORMAL_INSTALL}"
+  elif [ -e "${LOG4J_CONFIG_ZIP_INSTALL}" ]; then # Simple zip file layout
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:${LOG4J_CONFIG_ZIP_INSTALL}"
+  else # Fallback to normal default
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:$base_dir/../config/connect-log4j.properties"
+  fi
 fi
+export KAFKA_LOG4J_OPTS
 
 exec $(dirname $0)/kafka-run-class org.apache.kafka.connect.cli.ConnectStandalone "$@"
diff --git a/bin/kafka-run-class b/bin/kafka-run-class
index 1fb4eff..10dee75 100755
--- a/bin/kafka-run-class
+++ b/bin/kafka-run-class
@@ -108,7 +108,15 @@ fi
 # Log4j settings
 if [ -z "$KAFKA_LOG4J_OPTS" ]; then
   # Log to console. This is a tool.
-  KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:$base_dir/config/tools-log4j.properties"
+  LOG4J_CONFIG_NORMAL_INSTALL="/etc/kafka/tools-log4j.properties"
+  LOG4J_CONFIG_ZIP_INSTALL="$base_dir/etc/kafka/tools-log4j.properties"
+  if [ -e "$LOG4J_CONFIG_NORMAL_INSTALL" ]; then # Normal install layout
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:${LOG4J_CONFIG_NORMAL_INSTALL}"
+  elif [ -e "${LOG4J_CONFIG_ZIP_INSTALL}" ]; then # Simple zip file layout
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:${LOG4J_CONFIG_ZIP_INSTALL}"
+  else # Fallback to normal default
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:$base_dir/config/tools-log4j.properties"
+  fi
 else
   # create logs directory
   if [ ! -d "$LOG_DIR" ]; then
diff --git a/bin/kafka-server-start b/bin/kafka-server-start
index e74a0ae..5c364fb 100755
--- a/bin/kafka-server-start
+++ b/bin/kafka-server-start
@@ -22,8 +22,17 @@ fi
 base_dir=$(dirname $0)
 
 if [ "x$KAFKA_LOG4J_OPTS" = "x" ]; then
-    export KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:$base_dir/../config/log4j.properties"
+  LOG4J_CONFIG_NORMAL_INSTALL="/etc/kafka/log4j.properties"
+  LOG4J_CONFIG_ZIP_INSTALL="$base_dir/../etc/kafka/log4j.properties"
+  if [ -e "$LOG4J_CONFIG_NORMAL_INSTALL" ]; then # Normal install layout
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:${LOG4J_CONFIG_NORMAL_INSTALL}"
+  elif [ -e "${LOG4J_CONFIG_ZIP_INSTALL}" ]; then # Simple zip file layout
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:${LOG4J_CONFIG_ZIP_INSTALL}"
+  else # Fallback to normal default
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:$base_dir/../config/log4j.properties"
+  fi
 fi
+export KAFKA_LOG4J_OPTS
 
 if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
     export KAFKA_HEAP_OPTS="-Xmx1G -Xms1G"
diff --git a/bin/windows/kafka-run-class.bat b/bin/windows/kafka-run-class.bat
index f5b8faa..64dc42f 100644
--- a/bin/windows/kafka-run-class.bat
+++ b/bin/windows/kafka-run-class.bat
@@ -87,7 +87,11 @@ IF ["%JMX_PORT%"] NEQ [""] (
 
 rem Log4j settings
 IF ["%KAFKA_LOG4J_OPTS%"] EQU [""] (
-	set KAFKA_LOG4J_OPTS=-Dlog4j.configuration=file:%BASE_DIR%/config/tools-log4j.properties
+	if exists %~dp0../../etc/kafka/tools-log4j.properties (
+		set KAFKA_LOG4J_OPTS=-Dlog4j.configuration=file:%~dp0../../etc/kafka/tools-log4j.properties
+	) else (
+		set KAFKA_LOG4J_OPTS=-Dlog4j.configuration=file:%BASE_DIR%/config/tools-log4j.properties
+	)
 )
 
 rem Generic jvm settings you want to add
diff --git a/bin/windows/kafka-server-start.bat b/bin/windows/kafka-server-start.bat
index 6a5aa9f..ae56b48 100644
--- a/bin/windows/kafka-server-start.bat
+++ b/bin/windows/kafka-server-start.bat
@@ -20,7 +20,11 @@ IF [%1] EQU [] (
 )
 
 SetLocal
-set KAFKA_LOG4J_OPTS=-Dlog4j.configuration=file:%~dp0../../config/log4j.properties
+if exists %~dp0../../etc/kafka/log4j.properties (
+	set KAFKA_LOG4J_OPTS=-Dlog4j.configuration=file:%~dp0../../etc/kafka/log4j.properties
+) else (
+	set KAFKA_LOG4J_OPTS=-Dlog4j.configuration=file:%~dp0../../config/log4j.properties
+)
 set KAFKA_HEAP_OPTS=-Xmx1G -Xms1G
 %~dp0kafka-run-class.bat kafka.Kafka %*
 EndLocal
diff --git a/bin/windows/zookeeper-server-start.bat b/bin/windows/zookeeper-server-start.bat
index 9836283..9a31936 100644
--- a/bin/windows/zookeeper-server-start.bat
+++ b/bin/windows/zookeeper-server-start.bat
@@ -20,7 +20,11 @@ IF [%1] EQU [] (
 )
 
 SetLocal
-set KAFKA_LOG4J_OPTS=-Dlog4j.configuration=file:%~dp0../../config/log4j.properties
+if exists %~dp0../../etc/kafka/log4j.properties (
+	set KAFKA_LOG4J_OPTS=-Dlog4j.configuration=file:%~dp0../../etc/kafka/log4j.properties
+) else (
+	set KAFKA_LOG4J_OPTS=-Dlog4j.configuration=file:%~dp0../../config/log4j.properties
+)
 set KAFKA_HEAP_OPTS=-Xmx512M -Xms512M
 %~dp0kafka-run-class.bat org.apache.zookeeper.server.quorum.QuorumPeerMain %*
 EndLocal
diff --git a/bin/zookeeper-server-start b/bin/zookeeper-server-start
index fbbc8f6..755d753 100755
--- a/bin/zookeeper-server-start
+++ b/bin/zookeeper-server-start
@@ -22,8 +22,17 @@ fi
 base_dir=$(dirname $0)
 
 if [ "x$KAFKA_LOG4J_OPTS" = "x" ]; then
-    export KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:$base_dir/../config/log4j.properties"
+  LOG4J_CONFIG_NORMAL_INSTALL="/etc/kafka/log4j.properties"
+  LOG4J_CONFIG_ZIP_INSTALL="$base_dir/../etc/kafka/log4j.properties"
+  if [ -e "$LOG4J_CONFIG_NORMAL_INSTALL" ]; then # Normal install layout
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:${LOG4J_CONFIG_NORMAL_INSTALL}"
+  elif [ -e "${LOG4J_CONFIG_ZIP_INSTALL}" ]; then # Simple zip file layout
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:${LOG4J_CONFIG_ZIP_INSTALL}"
+  else # Fallback to normal default
+    KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:$base_dir/../config/log4j.properties"
+  fi
 fi
+export KAFKA_LOG4J_OPTS
 
 if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
     export KAFKA_HEAP_OPTS="-Xmx512M -Xms512M"
