diff --git a/build.gradle b/build.gradle
index c3e6bb8..331e5e6 100644
--- a/build.gradle
+++ b/build.gradle
@@ -128,8 +128,6 @@ subprojects {
   }
 
   tasks.withType(ScalaCompile) {
-    scalaCompileOptions.useAnt = false
-
     configure(scalaCompileOptions.forkOptions) {
       memoryMaximumSize = '1g'
       jvmArgs = ['-XX:MaxPermSize=512m']
@@ -223,8 +221,6 @@ project(':core') {
       testCompile "org.scalatest:scalatest_$scalaVersion:1.8"
     }
     testRuntime "$slf4jlog4j"
-
-    zinc 'com.typesafe.zinc:zinc:0.3.1'
   }
 
   configurations {
