diff --git a/build.gradle b/build.gradle
index 49733c4..2947113 100644
--- a/build.gradle
+++ b/build.gradle
@@ -179,8 +179,6 @@ subprojects {
   }
 
   tasks.withType(ScalaCompile) {
-    scalaCompileOptions.useAnt = false
-
     configure(scalaCompileOptions.forkOptions) {
       memoryMaximumSize = '1g'
       jvmArgs = ['-XX:MaxPermSize=512m', '-Xss2m']
@@ -274,7 +272,6 @@ project(':core') {
     testCompile project(':clients').sourceSets.test.output
     testCompile 'org.apache.hadoop:hadoop-minikdc:2.7.1'
     testRuntime "$slf4jlog4j"
-    zinc 'com.typesafe.zinc:zinc:0.3.7'
   }
 
   configurations {
