diff --git a/bin/connect-distributed b/bin/connect-distributed
index 65c5f8b..1ca1cc9 100755
--- a/bin/connect-distributed
+++ b/bin/connect-distributed
@@ -16,6 +16,19 @@
 
 base_dir=$(dirname $0)
 
+# classpath addition for any Kafka Connect connectors
+# included in releases with LSB-style layout
+for dir in $base_dir/../share/java/kafka-connect-*; do
+  CLASSPATH=$CLASSPATH:$dir/*
+done
+
+# classpath addition for kafka-serde-tools (e.g. Avro serializer),
+# which are currently bundled with the Confluent schema-registry package.
+# kafka-serde-tools also require confluent-common
+for library in "confluent-common" "kafka-serde-tools"; do
+  CLASSPATH=$CLASSPATH:$base_dir/../share/java/$library/*
+done
+
 if [ "x$KAFKA_LOG4J_OPTS" = "x" ]; then
   LOG4J_CONFIG_NORMAL_INSTALL="/etc/kafka/connect-log4j.properties"
   LOG4J_CONFIG_ZIP_INSTALL="$base_dir/../etc/kafka/connect-log4j.properties"
@@ -29,4 +42,5 @@ if [ "x$KAFKA_LOG4J_OPTS" = "x" ]; then
 fi
 export KAFKA_LOG4J_OPTS
 
+export CLASSPATH
 exec $(dirname $0)/kafka-run-class org.apache.kafka.connect.cli.ConnectDistributed "$@"
diff --git a/bin/connect-standalone b/bin/connect-standalone
index ca58fe6..b39bc88 100755
--- a/bin/connect-standalone
+++ b/bin/connect-standalone
@@ -16,6 +16,19 @@
 
 base_dir=$(dirname $0)
 
+# classpath addition for any Kafka Connect connectors
+# included in releases with LSB-style layout
+for dir in $base_dir/../share/java/kafka-connect-*; do
+  CLASSPATH=$CLASSPATH:$dir/*
+done
+
+# classpath addition for kafka-serde-tools (e.g. Avro serializer),
+# which are currently bundled with the Confluent schema-registry package.
+# kafka-serde-tools also require confluent-common
+for library in "confluent-common" "kafka-serde-tools"; do
+  CLASSPATH=$CLASSPATH:$base_dir/../share/java/$library/*
+done
+
 if [ "x$KAFKA_LOG4J_OPTS" = "x" ]; then
   LOG4J_CONFIG_NORMAL_INSTALL="/etc/kafka/connect-log4j.properties"
   LOG4J_CONFIG_ZIP_INSTALL="$base_dir/../etc/kafka/connect-log4j.properties"
@@ -29,4 +42,5 @@ if [ "x$KAFKA_LOG4J_OPTS" = "x" ]; then
 fi
 export KAFKA_LOG4J_OPTS
 
+export CLASSPATH
 exec $(dirname $0)/kafka-run-class org.apache.kafka.connect.cli.ConnectStandalone "$@"
