diff --git a/bin/kafka-run-class b/bin/kafka-run-class
index b18a9cf..21cc4b7 100755
--- a/bin/kafka-run-class
+++ b/bin/kafka-run-class
@@ -86,6 +86,9 @@ done
 # classpath addition for release
 CLASSPATH=$CLASSPATH:$base_dir/libs/*
 
+# classpath addition for releases with LSB-style layout
+CLASSPATH=$CLASSPATH:$base_dir/share/java/kafka/*
+
 for file in $base_dir/core/build/libs/kafka_${SCALA_BINARY_VERSION}*.jar;
 do
   CLASSPATH=$CLASSPATH:$file
