diff --git a/bin/kafka-run-class.sh b/bin/kafka-run-class.sh
index 36c742b..8c6991e 100755
--- a/bin/kafka-run-class
+++ b/bin/kafka-run-class
@@ -71,6 +71,12 @@ do
   CLASSPATH=$CLASSPATH:$file
 done
 
+# classpath addition for releases with LSB-style layout
+for file in $base_dir/share/java/kafka-0.8.2.2/*.jar;
+do
+  CLASSPATH=$CLASSPATH:$file
+done
+
 for file in $base_dir/core/build/libs/kafka_${SCALA_BINARY_VERSION}*.jar;
 do
   CLASSPATH=$CLASSPATH:$file
