diff --git a/bin/kafka-run-class b/bin/kafka-run-class
index d98533d..7e58ac3 100755
--- a/bin/kafka-run-class
+++ b/bin/kafka-run-class
@@ -85,6 +85,12 @@ do
   CLASSPATH=$CLASSPATH:$file
 done
 
+# classpath addition for releases with LSB-style layout
+for file in $base_dir/share/java/kafka/*.jar;
+do
+  CLASSPATH=$CLASSPATH:$file
+done
+
 for file in $base_dir/core/build/libs/kafka_${SCALA_BINARY_VERSION}*.jar;
 do
   CLASSPATH=$CLASSPATH:$file
