diff --git a/bin/connect-distributed.sh b/bin/connect-distributed
similarity index 92%
rename from bin/connect-distributed.sh
rename to bin/connect-distributed
index 99cd27b..4dad66a 100755
--- a/bin/connect-distributed.sh
+++ b/bin/connect-distributed
@@ -38,4 +38,4 @@ case $COMMAND in
     ;;
 esac
 
-exec $(dirname $0)/kafka-run-class.sh $EXTRA_ARGS org.apache.kafka.connect.cli.ConnectDistributed "$@"
+exec $(dirname $0)/kafka-run-class $EXTRA_ARGS org.apache.kafka.connect.cli.ConnectDistributed "$@"
diff --git a/bin/connect-standalone.sh b/bin/connect-standalone
similarity index 92%
rename from bin/connect-standalone.sh
rename to bin/connect-standalone
index 623562a..5c8be55 100755
--- a/bin/connect-standalone.sh
+++ b/bin/connect-standalone
@@ -38,4 +38,4 @@ case $COMMAND in
     ;;
 esac
 
-exec $(dirname $0)/kafka-run-class.sh $EXTRA_ARGS org.apache.kafka.connect.cli.ConnectStandalone "$@"
+exec $(dirname $0)/kafka-run-class $EXTRA_ARGS org.apache.kafka.connect.cli.ConnectStandalone "$@"
diff --git a/bin/kafka-mirror-maker.sh b/bin/kafka-acls
similarity index 92%
rename from bin/kafka-mirror-maker.sh
rename to bin/kafka-acls
index 981f271..72c0629 100755
--- a/bin/kafka-mirror-maker.sh
+++ b/bin/kafka-acls
@@ -14,4 +14,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-exec $(dirname $0)/kafka-run-class.sh kafka.tools.MirrorMaker "$@"
+exec $(dirname $0)/kafka-run-class kafka.admin.AclCommand "$@"
diff --git a/bin/kafka-acls.sh b/bin/kafka-configs
similarity index 92%
rename from bin/kafka-acls.sh
rename to bin/kafka-configs
index 8fa6554..35b5dd4 100755
--- a/bin/kafka-acls.sh
+++ b/bin/kafka-configs
@@ -14,4 +14,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-exec $(dirname $0)/kafka-run-class.sh kafka.admin.AclCommand "$@"
+exec $(dirname $0)/kafka-run-class kafka.admin.ConfigCommand "$@"
diff --git a/bin/kafka-console-consumer.sh b/bin/kafka-console-consumer
similarity index 92%
rename from bin/kafka-console-consumer.sh
rename to bin/kafka-console-consumer
index dbaac2b..01dfa76 100755
--- a/bin/kafka-console-consumer.sh
+++ b/bin/kafka-console-consumer
@@ -18,4 +18,4 @@ if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
     export KAFKA_HEAP_OPTS="-Xmx512M"
 fi
 
-exec $(dirname $0)/kafka-run-class.sh kafka.tools.ConsoleConsumer "$@"
+exec $(dirname $0)/kafka-run-class kafka.tools.ConsoleConsumer "$@"
diff --git a/bin/kafka-console-producer.sh b/bin/kafka-console-producer
similarity index 92%
rename from bin/kafka-console-producer.sh
rename to bin/kafka-console-producer
index e5187b8..11642e7 100755
--- a/bin/kafka-console-producer.sh
+++ b/bin/kafka-console-producer
@@ -17,4 +17,4 @@
 if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
     export KAFKA_HEAP_OPTS="-Xmx512M"
 fi
-exec $(dirname $0)/kafka-run-class.sh kafka.tools.ConsoleProducer "$@"
+exec $(dirname $0)/kafka-run-class kafka.tools.ConsoleProducer "$@"
diff --git a/bin/kafka-replay-log-producer.sh b/bin/kafka-consumer-groups
similarity index 91%
rename from bin/kafka-replay-log-producer.sh
rename to bin/kafka-consumer-groups
index bba3241..b902db6 100755
--- a/bin/kafka-replay-log-producer.sh
+++ b/bin/kafka-consumer-groups
@@ -14,4 +14,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-exec $(dirname $0)/kafka-run-class.sh kafka.tools.ReplayLogProducer "$@"
+exec $(dirname $0)/kafka-run-class kafka.admin.ConsumerGroupCommand "$@"
diff --git a/bin/kafka-consumer-offset-checker b/bin/kafka-consumer-offset-checker
new file mode 100755
index 0000000..e6ad445
--- /dev/null
+++ b/bin/kafka-consumer-offset-checker
@@ -0,0 +1,17 @@
+#!/bin/bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+# 
+#    http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+exec $(dirname $0)/kafka-run-class kafka.tools.ConsumerOffsetChecker "$@"
diff --git a/bin/kafka-consumer-offset-checker.sh b/bin/kafka-consumer-offset-checker.sh
deleted file mode 100755
index 5993345..0000000
--- a/bin/kafka-consumer-offset-checker.sh
+++ /dev/null
@@ -1,17 +0,0 @@
-#!/bin/bash
-# Licensed to the Apache Software Foundation (ASF) under one or more
-# contributor license agreements.  See the NOTICE file distributed with
-# this work for additional information regarding copyright ownership.
-# The ASF licenses this file to You under the Apache License, Version 2.0
-# (the "License"); you may not use this file except in compliance with
-# the License.  You may obtain a copy of the License at
-# 
-#    http://www.apache.org/licenses/LICENSE-2.0
-# 
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-exec $(dirname $0)/kafka-run-class.sh kafka.tools.ConsumerOffsetChecker "$@"
diff --git a/bin/kafka-consumer-perf-test.sh b/bin/kafka-consumer-perf-test
similarity index 92%
rename from bin/kafka-consumer-perf-test.sh
rename to bin/kafka-consumer-perf-test
index 77cda72..3b1fef8 100755
--- a/bin/kafka-consumer-perf-test.sh
+++ b/bin/kafka-consumer-perf-test
@@ -17,4 +17,4 @@
 if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
     export KAFKA_HEAP_OPTS="-Xmx512M"
 fi
-exec $(dirname $0)/kafka-run-class.sh kafka.tools.ConsumerPerformance "$@"
+exec $(dirname $0)/kafka-run-class kafka.tools.ConsumerPerformance "$@"
diff --git a/bin/kafka-topics.sh b/bin/kafka-mirror-maker
similarity index 92%
rename from bin/kafka-topics.sh
rename to bin/kafka-mirror-maker
index ad6a2d4..e5626e5 100755
--- a/bin/kafka-topics.sh
+++ b/bin/kafka-mirror-maker
@@ -14,4 +14,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-exec $(dirname $0)/kafka-run-class.sh kafka.admin.TopicCommand "$@"
+exec $(dirname $0)/kafka-run-class kafka.tools.MirrorMaker "$@"
diff --git a/bin/kafka-preferred-replica-election.sh b/bin/kafka-preferred-replica-election
similarity index 89%
rename from bin/kafka-preferred-replica-election.sh
rename to bin/kafka-preferred-replica-election
index 638a92a..495a833 100755
--- a/bin/kafka-preferred-replica-election.sh
+++ b/bin/kafka-preferred-replica-election
@@ -14,4 +14,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-exec $(dirname $0)/kafka-run-class.sh kafka.admin.PreferredReplicaLeaderElectionCommand "$@"
+exec $(dirname $0)/kafka-run-class kafka.admin.PreferredReplicaLeaderElectionCommand "$@"
diff --git a/bin/kafka-producer-perf-test b/bin/kafka-producer-perf-test
new file mode 100755
index 0000000..43d7965
--- /dev/null
+++ b/bin/kafka-producer-perf-test
@@ -0,0 +1,20 @@
+#!/bin/bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
+    export KAFKA_HEAP_OPTS="-Xmx512M"
+fi
+exec $(dirname $0)/kafka-run-class org.apache.kafka.tools.ProducerPerformance "$@"
diff --git a/bin/kafka-producer-perf-test.sh b/bin/kafka-producer-perf-test.sh
deleted file mode 100755
index 73a6288..0000000
--- a/bin/kafka-producer-perf-test.sh
+++ /dev/null
@@ -1,20 +0,0 @@
-#!/bin/bash
-# Licensed to the Apache Software Foundation (ASF) under one or more
-# contributor license agreements.  See the NOTICE file distributed with
-# this work for additional information regarding copyright ownership.
-# The ASF licenses this file to You under the Apache License, Version 2.0
-# (the "License"); you may not use this file except in compliance with
-# the License.  You may obtain a copy of the License at
-#
-#    http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
-    export KAFKA_HEAP_OPTS="-Xmx512M"
-fi
-exec $(dirname $0)/kafka-run-class.sh org.apache.kafka.tools.ProducerPerformance "$@"
diff --git a/bin/kafka-reassign-partitions.sh b/bin/kafka-reassign-partitions
similarity index 90%
rename from bin/kafka-reassign-partitions.sh
rename to bin/kafka-reassign-partitions
index 4c7f1bc..af40ced 100755
--- a/bin/kafka-reassign-partitions.sh
+++ b/bin/kafka-reassign-partitions
@@ -14,4 +14,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-exec $(dirname $0)/kafka-run-class.sh kafka.admin.ReassignPartitionsCommand "$@"
+exec $(dirname $0)/kafka-run-class kafka.admin.ReassignPartitionsCommand "$@"
diff --git a/bin/kafka-replay-log-producer b/bin/kafka-replay-log-producer
new file mode 100755
index 0000000..eefc24d
--- /dev/null
+++ b/bin/kafka-replay-log-producer
@@ -0,0 +1,17 @@
+#!/bin/bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+# 
+#    http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+exec $(dirname $0)/kafka-run-class kafka.tools.ReplayLogProducer "$@"
diff --git a/bin/kafka-consumer-groups.sh b/bin/kafka-replica-verification
similarity index 91%
rename from bin/kafka-consumer-groups.sh
rename to bin/kafka-replica-verification
index feb063d..77689fe 100755
--- a/bin/kafka-consumer-groups.sh
+++ b/bin/kafka-replica-verification
@@ -14,4 +14,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-exec $(dirname $0)/kafka-run-class.sh kafka.admin.ConsumerGroupCommand "$@"
+exec $(dirname $0)/kafka-run-class kafka.tools.ReplicaVerificationTool "$@"
diff --git a/bin/kafka-replica-verification.sh b/bin/kafka-replica-verification.sh
deleted file mode 100755
index 4960836..0000000
--- a/bin/kafka-replica-verification.sh
+++ /dev/null
@@ -1,17 +0,0 @@
-#!/bin/bash
-# Licensed to the Apache Software Foundation (ASF) under one or more
-# contributor license agreements.  See the NOTICE file distributed with
-# this work for additional information regarding copyright ownership.
-# The ASF licenses this file to You under the Apache License, Version 2.0
-# (the "License"); you may not use this file except in compliance with
-# the License.  You may obtain a copy of the License at
-# 
-#    http://www.apache.org/licenses/LICENSE-2.0
-# 
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-exec $(dirname $0)/kafka-run-class.sh kafka.tools.ReplicaVerificationTool "$@"
diff --git a/bin/kafka-run-class.sh b/bin/kafka-run-class
similarity index 100%
rename from bin/kafka-run-class.sh
rename to bin/kafka-run-class
diff --git a/bin/kafka-server-start.sh b/bin/kafka-server-start
similarity index 95%
rename from bin/kafka-server-start.sh
rename to bin/kafka-server-start
index 5a53126..e1dcb63 100755
--- a/bin/kafka-server-start.sh
+++ b/bin/kafka-server-start
@@ -41,4 +41,4 @@ case $COMMAND in
     ;;
 esac
 
-exec $base_dir/kafka-run-class.sh $EXTRA_ARGS kafka.Kafka "$@"
+exec $base_dir/kafka-run-class $EXTRA_ARGS kafka.Kafka "$@"
diff --git a/bin/kafka-server-stop.sh b/bin/kafka-server-stop
similarity index 100%
rename from bin/kafka-server-stop.sh
rename to bin/kafka-server-stop
diff --git a/bin/kafka-simple-consumer-shell b/bin/kafka-simple-consumer-shell
new file mode 100755
index 0000000..fd7232a
--- /dev/null
+++ b/bin/kafka-simple-consumer-shell
@@ -0,0 +1,17 @@
+#!/bin/bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+# 
+#    http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+exec $(dirname $0)/kafka-run-class kafka.tools.SimpleConsumerShell "$@"
diff --git a/bin/kafka-simple-consumer-shell.sh b/bin/kafka-simple-consumer-shell.sh
deleted file mode 100755
index 27e386a..0000000
--- a/bin/kafka-simple-consumer-shell.sh
+++ /dev/null
@@ -1,17 +0,0 @@
-#!/bin/bash
-# Licensed to the Apache Software Foundation (ASF) under one or more
-# contributor license agreements.  See the NOTICE file distributed with
-# this work for additional information regarding copyright ownership.
-# The ASF licenses this file to You under the Apache License, Version 2.0
-# (the "License"); you may not use this file except in compliance with
-# the License.  You may obtain a copy of the License at
-# 
-#    http://www.apache.org/licenses/LICENSE-2.0
-# 
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-exec $(dirname $0)/kafka-run-class.sh kafka.tools.SimpleConsumerShell "$@"
diff --git a/bin/kafka-streams-application-reset.sh b/bin/kafka-streams-application-reset
similarity index 92%
rename from bin/kafka-streams-application-reset.sh
rename to bin/kafka-streams-application-reset
index 3363732..c9cde3f 100755
--- a/bin/kafka-streams-application-reset.sh
+++ b/bin/kafka-streams-application-reset
@@ -18,4 +18,4 @@ if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
     export KAFKA_HEAP_OPTS="-Xmx512M"
 fi
 
-exec $(dirname $0)/kafka-run-class.sh kafka.tools.StreamsResetter "$@"
+exec $(dirname $0)/kafka-run-class kafka.tools.StreamsResetter "$@"
diff --git a/bin/kafka-configs.sh b/bin/kafka-topics
similarity index 92%
rename from bin/kafka-configs.sh
rename to bin/kafka-topics
index 2f9eb8c..cb39c54 100755
--- a/bin/kafka-configs.sh
+++ b/bin/kafka-topics
@@ -14,4 +14,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-exec $(dirname $0)/kafka-run-class.sh kafka.admin.ConfigCommand "$@"
+exec $(dirname $0)/kafka-run-class kafka.admin.TopicCommand "$@"
diff --git a/bin/kafka-verifiable-consumer b/bin/kafka-verifiable-consumer
new file mode 100755
index 0000000..29b307c
--- /dev/null
+++ b/bin/kafka-verifiable-consumer
@@ -0,0 +1,20 @@
+#!/bin/bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
+    export KAFKA_HEAP_OPTS="-Xmx512M"
+fi
+exec $(dirname $0)/kafka-run-class org.apache.kafka.tools.VerifiableConsumer "$@"
diff --git a/bin/kafka-verifiable-consumer.sh b/bin/kafka-verifiable-consumer.sh
deleted file mode 100755
index 852847d..0000000
--- a/bin/kafka-verifiable-consumer.sh
+++ /dev/null
@@ -1,20 +0,0 @@
-#!/bin/bash
-# Licensed to the Apache Software Foundation (ASF) under one or more
-# contributor license agreements.  See the NOTICE file distributed with
-# this work for additional information regarding copyright ownership.
-# The ASF licenses this file to You under the Apache License, Version 2.0
-# (the "License"); you may not use this file except in compliance with
-# the License.  You may obtain a copy of the License at
-#
-#    http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
-    export KAFKA_HEAP_OPTS="-Xmx512M"
-fi
-exec $(dirname $0)/kafka-run-class.sh org.apache.kafka.tools.VerifiableConsumer "$@"
diff --git a/bin/kafka-verifiable-producer b/bin/kafka-verifiable-producer
new file mode 100755
index 0000000..40bc910
--- /dev/null
+++ b/bin/kafka-verifiable-producer
@@ -0,0 +1,20 @@
+#!/bin/bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
+    export KAFKA_HEAP_OPTS="-Xmx512M"
+fi
+exec $(dirname $0)/kafka-run-class org.apache.kafka.tools.VerifiableProducer "$@"
diff --git a/bin/kafka-verifiable-producer.sh b/bin/kafka-verifiable-producer.sh
deleted file mode 100755
index b59bae7..0000000
--- a/bin/kafka-verifiable-producer.sh
+++ /dev/null
@@ -1,20 +0,0 @@
-#!/bin/bash
-# Licensed to the Apache Software Foundation (ASF) under one or more
-# contributor license agreements.  See the NOTICE file distributed with
-# this work for additional information regarding copyright ownership.
-# The ASF licenses this file to You under the Apache License, Version 2.0
-# (the "License"); you may not use this file except in compliance with
-# the License.  You may obtain a copy of the License at
-#
-#    http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-if [ "x$KAFKA_HEAP_OPTS" = "x" ]; then
-    export KAFKA_HEAP_OPTS="-Xmx512M"
-fi
-exec $(dirname $0)/kafka-run-class.sh org.apache.kafka.tools.VerifiableProducer "$@"
diff --git a/bin/zookeeper-security-migration.sh b/bin/zookeeper-security-migration
similarity index 91%
rename from bin/zookeeper-security-migration.sh
rename to bin/zookeeper-security-migration
index 722bde7..a43c212 100755
--- a/bin/zookeeper-security-migration.sh
+++ b/bin/zookeeper-security-migration
@@ -14,4 +14,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-exec $(dirname $0)/kafka-run-class.sh kafka.admin.ZkSecurityMigrator "$@"
+exec $(dirname $0)/kafka-run-class kafka.admin.ZkSecurityMigrator "$@"
diff --git a/bin/zookeeper-server-start.sh b/bin/zookeeper-server-start
similarity index 92%
rename from bin/zookeeper-server-start.sh
rename to bin/zookeeper-server-start
index bd9c114..9ad07af 100755
--- a/bin/zookeeper-server-start.sh
+++ b/bin/zookeeper-server-start
@@ -41,4 +41,4 @@ case $COMMAND in
      ;;
 esac
 
-exec $base_dir/kafka-run-class.sh $EXTRA_ARGS org.apache.zookeeper.server.quorum.QuorumPeerMain "$@"
+exec $base_dir/kafka-run-class $EXTRA_ARGS org.apache.zookeeper.server.quorum.QuorumPeerMain "$@"
diff --git a/bin/zookeeper-server-stop.sh b/bin/zookeeper-server-stop
similarity index 100%
rename from bin/zookeeper-server-stop.sh
rename to bin/zookeeper-server-stop
diff --git a/bin/zookeeper-shell.sh b/bin/zookeeper-shell
similarity index 91%
rename from bin/zookeeper-shell.sh
rename to bin/zookeeper-shell
index 95007fa..6860532 100755
--- a/bin/zookeeper-shell.sh
+++ b/bin/zookeeper-shell
@@ -20,4 +20,4 @@ then
 	exit 1
 fi
 
-exec $(dirname $0)/kafka-run-class.sh org.apache.zookeeper.ZooKeeperMain -server "$@"
+exec $(dirname $0)/kafka-run-class org.apache.zookeeper.ZooKeeperMain -server "$@"
